src/async/Task.ts

```diff
@@ -10,7 +10,7 @@ export const Canceled = <State> 4;
  * @param value The value to guard
  */
 export function isTask<T>(value: any): value is Task<T> {
-	return Boolean(isThenable(value) && typeof value.cancel === 'function' && Array.isArray(value.children));
+	return Boolean(isThenable(value) && typeof (<any> value).cancel === 'function' && Array.isArray((<any> value).children));
```

Comment:
I think the `isThenable` type guard is narrowing your `any` to a `Thenable`, so what you want to do is either reorder it or replicate the logic of the guard in `isThenable` without using it.